tests/sdp-user-extension.rs

```diff
@@ -0,0 +1,158 @@
+use std::net::Ipv4Addr;
+use std::time::Duration;
+
+use str0m::format::Codec;
+use str0m::media::{Direction, MediaKind};
+use str0m::rtp::Extension;
+use str0m::rtp::ExtensionSerializer;
+use str0m::rtp::ExtensionValues;
+use str0m::Rtc;
+use str0m::{Candidate, Event, RtcError};
+use tracing::info_span;
+
+mod common;
+use common::{init_log, progress, TestRtc};
+
+#[test]
+pub fn sdp_user_extension() -> Result<(), RtcError> {
+    init_log();
+
+    #[derive(Debug, PartialEq, Eq)]
+    struct MyValue(u16);
+
+    #[derive(Debug)]
+    struct MyValueSerializer;
+
+    impl ExtensionSerializer for MyValueSerializer {
+        fn write_to(&self, buf: &mut [u8], ev: &ExtensionValues) -> usize {
+            // Does output have space?
+            if buf.len() < 2 {
+                return 0;
```

Comment:
I don't think this should fail silently, we could return `Result` instead

Reply:
Okay. I wasn't thinking we need to bubble the error necessarily, could just convert it to a `WARN`